packages/api/src/services/event.service.ts

```diff
@@ -7,38 +7,38 @@ import {Service} from './service';
  * to be caught by any component in any module.
  */
 export class EventService implements Service {
-    /**
+  /**
      * Emits a {@link CustomEvent} of type passed <code>event.NAME</code> and detail <code>event.payload</code>.
      *
      * @param event - the event to be emitted.
      * @return the emitted event.
      */
-    emit(event: Event): CustomEvent {
-        const customEvent = new CustomEvent(event.NAME, {detail: event.payload});
-        this.getHostElement().dispatchEvent(customEvent);
-        return customEvent;
-    }
+  emit<T extends {} | undefined>(event: Event<T>): CustomEvent {
```

Comment:
Isn't it better the generic types to be defined on the class instead of on each separate method?

Reply:
it's more flexible on the methods